### PR TITLE
Fix #1178 - Remove Kogito Operator pipelines and references

### LINKED ISSUE #1178: Remove Kogito Operator Pipelines
See https://github.com/apache/incubator-kie-kogito-operator/issues/1537

### PR DESCRIPTION
**Pull request by ricardozanini**
See https://github.com/apache/incubator-kie-kogito-operator/issues/1537

I've kept the node and the image that builds the operator 'cause we reuse it for the SonataFlow operator too.

Also, the example-images pipeline CAN be disabled since we only use it for the Kogito Operator BDD. But we need a new assessment on it to understand if they are being used somewhere else. We can open another PR to take care of it.

### CODE REVIEW

**Comment by jstastny-cz** [comment]
@ricardozanini IMHO the example images were exclusively for operator BDD tests. Let's ask @radtriste to confirm and we could get rid of it already.

File: kogito-jenkins-node/image.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: quay.io/kiegroup/kogito-jenkins-node
 version: "1.40.0"
 from: quay.io/openshift/origin-jenkins-agent-base:4.14
-description: Image used by dynamic jenkins agent to build kogito-cloud-operator using podman
+description: Image used by dynamic jenkins agent to build Kogito Operators using podman
```
Comment (jstastny-cz) [inline kogito-jenkins-node/image.yaml]: I don't think we use this image in apache CI at all ....can you double-check? Or is that used elsewhere?

**Comment by radtriste** [comment]
As far as I remember, they were mostly used for the bdd tests. We were indeed pushing them but I don't think that we were mentioning them into the documentation and/or examples' readmes.
That should be easy to check (look for 'quay.io/kiegroup/example-*' string

**Review by radtriste — APPROVED** [review]

File: kogito-jenkins-node/image.yaml (hunk 1 above)
Comment (radtriste) [reply kogito-jenkins-node/image.yaml]: I think the `kogito-jenkins-node` can be deleted. It was used in IBM Jenkins for external access. But now we are using Apache Jenkins

**Review by rodrigonull — APPROVED** [review]
LGTM.

**Comment by ricardozanini** [comment]
@radtriste @jstastny-cz many thanks for the updates! I'll remove the Jenkins Node and the BDD tests. I'll also squash this commit to make it easier to grab anything back if needed.

**Comment by ricardozanini** [comment]
@jstastny-cz @radtriste can you please take a look to make sure that my exclusions are alright? Thank you!

**Review by radtriste — APPROVED** [review]

**Review by jstastny-cz — APPROVED** [review]

**Pull request merged**